run.c

```diff
@@ -58,14 +59,50 @@ void run(void)
          * If needed, this function will also write an output file
          * at the desired time.
          */
-        int NumForces = find_next_sync_point(All.Ti_nextoutput);
+        All.Ti_Current = find_next_sync_point(All.Ti_nextoutput);
+
+        /* drift and domain decomposition */
+        /* at first step this is a noop */
+        if(All.PM_Ti_endstep == All.Ti_Current) {
+            /* PM step requires synced positions of all particles */
+
+            drift_all_particles(All.Ti_Current);
+
+            /* full decomposition rebuilds the tree */
+            domain_Decomposition();
+
+        } else {
+
+            drift_active_particles(All.Ti_Current);
```

Comment:
Is it ok to only drift active particles here? I am thinking of SPH, but I guess that is handled by VelPred?